CsvParser/csvParser_test.go

```diff
@@ -104,29 +164,141 @@ func Test_CsvParserFiftyFirstEntry(t *testing.T) {
 	}
 }
 
-func Test_MakeJsonObjFile(t *testing.T) {
+func Test_FileHierachy(t *testing.T) {
+	ex, err := os.Executable()
+	if err != nil {
+		panic(err)
+	}
+	originPath := filepath.Dir(ex)
+
+	incidentMap := mapIncidentsToLocAndISP(generateTestIncidents())
 
-	testIncidentArr := make([]incident.Incident, 1, 1)
-	incidents := CsvParser("incidentfile.csv")
-	testIncident := convertDefaultIncidentToIncident(incidents)
-	testIncidentArr[0] = testIncident[0]
+	locationCodesArr := make([]string, 0)
+	locationCodesArr = append(locationCodesArr, "/eu/fr")
+	locationCodesArr = append(locationCodesArr, "/eu/fr/AS10774x.json")
+	locationCodesArr = append(locationCodesArr, "/eu/fr/AS10774z.json")
+	locationCodesArr = append(locationCodesArr, "/na/us")
 
+	tests := []struct {
+		name        string
+		input  []string
+	}{
+		{
+			name: "Check of the directories are being created",
+			input: locationCodesArr,
+		},
+	}
+
+	placeIncidentsInFileHierarchy(originPath, incidentMap)
+	for _, tt := range tests {
+
+		i := 0
+
+		t.Run(tt.name, func(t *testing.T) {
+			for ( i != 4) {
+				if _, err := os.Stat(originPath + tt.input[i]); os.IsNotExist(err) {
+					t.Errorf("File does not exist")
+				}
+				i++
+			}
+		})
+		
+	}
+}
+
+func Test_ParseLocationCode(t *testing.T) {
+	locationCodesArr := make([]string, 0)
+	locationCodesArr = append(locationCodesArr, "na")
+	locationCodesArr = append(locationCodesArr, "us")
+	locationCodesArr = append(locationCodesArr, "wa")
+	locationCodesArr = append(locationCodesArr, "redmond")
+	
 	tests := []struct {
 		name  string
-		input []incident.Incident
+		input string
+		want  []string
 	}{
 		{
-			name:  "Create a file with one entry",
-			input: testIncidentArr,
+			name:  "Parsing the location code string",
+			input: "nauswaredmond",
+			want: locationCodesArr,
 		},
 	}
 
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
-			makeJsonObjFile(tt.input)
-			if _, err := os.Stat("incidents.json"); os.IsNotExist(err) {
-				t.Errorf("File does not exist")
+			parsedCode := parseLocationCode(tt.input)
+			if !reflect.DeepEqual(parsedCode, tt.want) {
+				t.Errorf("parseLocationCode() = %v, want %v", parsedCode, tt.want)
 			}
 		})
 	}
+
 }
+
+func Test_incidentsMemPlacer(t *testing.T) {
+    
+    tests := []struct {
+        name  string
+		input string
+		inputTwo string
+        want  string
+    }{
+        {
+            name:  "A location that should exist in the map",
+            input: "AS10774x",
+            inputTwo: "naus",
+            want:  "naus",
+        },
+
+        {
+            name:  "A location that has two ISPs associated w/ it",
+            input: "AS10774z",
+            inputTwo: "eufr",
+            want:  "eufr",
+        },
+
+        {
+            name:  "A location that has two ISPs associated w/ it",
+            input: "AS10774x",
```

Comment:
Maybe these tests could be more readable if input was changed to "asn" and inputTwo was changed to "locationCode"

Reply:
I tried to be consistent with the "input" idiom

Reply:
Your preference... you can mark this convo as resolved if you decide not to change it